File: CodeRun_Boost/7268_choose_primes/solution.py
def solution(n: int) -> int:
    if n < 5:
        return 0

    is_prime = [True] * (n + 1)
    is_prime[0] = is_prime[1] = False
    for i in range(2, int(n**0.5) + 1):
        if is_prime[i]:
            # Mark all multiples of i as not prime.
            for multiple in range(i * i, n + 1, i):
                is_prime[multiple] = False

    count1_mod4 = 0
    count3_mod4 = 0
    for p in range(3, n + 1):
        if is_prime[p]:
            if p % 4 == 1:
                count1_mod4 += 1
            elif p % 4 == 3:
                count3_mod4 += 1
    return count1_mod4 * count3_mod4

File: CodeRun_Boost/7268_choose_primes/solution.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <math.h>

long long solution(int n) {
    if (n < 5) {
        return 0;
    }

    bool* is_prime = (bool*)malloc((n + 1) * sizeof(bool));
    for (int i = 0; i <= n; i++) {
        is_prime[i] = true;
    }
    is_prime[0] = is_prime[1] = false;

    for (int p = 2; p * p <= n; p++) {
        if (is_prime[p]) {
            for (int i = p * p; i <= n; i += p) {
                is_prime[i] = false;
            }
        }
    }

    long long count1_mod4 = 0;
    long long count3_mod4 = 0;

    for (int p = 3; p <= n; p++) {
        if (is_prime[p]) {
            if (p % 4 == 1) {
                count1_mod4++;
            } else if (p % 4 == 3) {
                count3_mod4++;
            }
        }
    }

    free(is_prime);

    return count1_mod4 * count3_mod4;
}